src/components/Staff/StaffTimeline.tsx

```diff
@@ -94,20 +74,82 @@ const StaffTimeline = ({ staffId, facilityId, systemId }: Props) => {
     [data]
   );
 
+  const filterSelectOptions: Option[] = [
+    {
+      label: "All Facilities",
+      value: "facilities",
+      children: data?.system.facilities.map((f) => ({
+        label: f.publicId,
+        value: f.id,
+      })),
+    },
+
+    {
+      label: "All Categories",
+      value: "eventTypes",
+      children: [
+        { label: "Connection", value: "Connection" },
+        { label: "Connection Restriction", value: "ConnectionRestriction" },
+        { label: "Meeting", value: "Meeting" },
+        { label: "Message", value: "Message" },
+        { label: "Message Keyword", value: "MessageKeyword" },
+        { label: "Kiosk", value: "Kiosk" },
+        { label: "Staff Position", value: "StaffPosition" },
+        { label: "Staff Entitlement", value: "StaffEntitlement" },
+      ],
+    },
+  ];
+
   const [filters, setFilters] = useState<Filters>({
     facilities: facilityId ? [facilityId] : [],
+    eventTypes: [],
   });
+
+  const [selections, setSelections] = useState<string[][]>(
+    facilityId ? [["facilities", facilityId]] : []
+  );
+
+  useEffect(() => {
+    // if a parent (All facilities or All eventTypes) is selected, it appears as ['facilities'] or ['eventTypes']
+    // if some but not all children of a parent are selected, they each appear as ['facilities', facilityId]
+    setFilters({
+      facilities:
+        selections.some(
+          (selection) => selection.length === 1 && selection[0] === "facilities"
+        ) && data
+          ? data?.system.facilities.map((f) => f.id) // If the parent is selected, include all facility ids
+          : selections
+              .filter(
+                (entry) => entry.length === 2 && entry[0] === "facilities"
+              )
+              .map((entry) => entry[1]),
+      eventTypes: selections.some(
+        (selection) => selection.length === 1 && selection[0] === "eventTypes"
+      )
+        ? filterSelectOptions[1].children?.map((c) => c.value as string) || [] // If the parent is selected, include all eventTypes
```

Comment:
would this also work given the `setFilteredTimeline` logic below?
```suggestion
        ? []
```